src/middlewares/auth.middleware.ts

```diff
@@ -0,0 +1,56 @@
+import {Request, Response, NextFunction} from "express";
+import {JWT} from "@/config";
+import jwt from "jsonwebtoken";
+
+declare module "express-serve-static-core" {
+    interface Request {
+        user?: {id: number, role: string, email?: string}
```

Comment:
use lab staff. from the design of this system. it does not appear any other group of people (e.g customer) will have access to it. so this object (user) at this  point should not be optional (?) every request that gets to the controller is expected to be authenticated and thus have this object at that point. authenticated requests dies at the middleware